neurons/validator.py

```diff
@@ -758,741 +758,177 @@ async def run(self):
-                tplr.r2_dataset.R2DatasetLoader.create,
-                batch_size=self.hparams.batch_size,
-                sequence_length=self.hparams.sequence_length,
-                pages_info=pages_random,
-                tokenizer=self.tokenizer,
-                attempts=3,
-                delay=1,
-                context="random loader - common",
-                **{},
+            # Compute offset for all data loading using the current sync_window and hparams
+            offset = self.sync_window * self.hparams.pages_per_window
+
+            from tplr import evaluation
+            eval_results, aggregated_metrics = await evaluation.evaluate_peers_parallel(
+                evaluation_uids,
+                self.comms,
+                self.sync_window,
+                self.hparams,
+                self.tokenizer,
+                self.config,
+                self.model,
+                self.transformer,
+                self.compressor,
+                self.xshapes,
+                self.totalks,
+                self.config.device,
+                self.optimizer.param_groups[0]["lr"],
+                self.optimizer,
+                self.scheduler,
+                time_min,
+                time_max
             )
-            tplr.logger.info(
-                f"{tplr.P(self.sync_window, tplr.T() - data_start_random)} Loaded common random loader for evaluation."
-            )
-
-            for eval_uid in evaluation_uids:
-                tplr.logger.info(f"Evaluating uid: {eval_uid}")
-
-                eval_result = await self.comms.get(
-                    uid=str(eval_uid),
-                    window=self.sync_window,
-                    key="gradient",
-                    local=False,
-                    stale_retention=10,
-                    time_max=time_max,
-                    time_min=time_min,
-                )
-
-                scoring_start = tplr.T()
-                if (
-                    eval_result is not None
-                    and not (
-                        isinstance(eval_result, dict)
-                        and eval_result.get("__status") in ["TOO_LATE", "TOO_EARLY"]
-                    )
-                    and eval_result[0] is not None
-                ):
-                    state_dict, _ = eval_result
-
-                    # Pull miner-sent pages info from metadata
-                    miner_pages = None
-                    if (
-                        "metadata" in state_dict
-                        and "pages_info" in state_dict["metadata"]
-                    ):
-                        miner_pages = state_dict["metadata"]["pages_info"]
-                    else:
-                        tplr.logger.warning(
-                            f"Missing pages info metadata from miner UID {eval_uid}"
+            # Process evaluation results and update scores (old evaluation logic)
+            for uid in evaluation_uids:
```

Comment:
i'd suggest we keep the `eval_uid` as it removes the confusion we're not referring to the validator's `uid`? wdyt?